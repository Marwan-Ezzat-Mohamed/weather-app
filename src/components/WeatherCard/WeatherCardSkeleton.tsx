const WeatherCardSkeleton = () => (
  <div className="min-w-sm relative flex w-full max-w-sm animate-pulse justify-center rounded-2xl bg-gray-300 p-5 dark:bg-gray-600">
    <div className="w-full space-y-5">
      <div className="flex justify-between">
        <span className="h-7 w-2/3 animate-pulse rounded bg-gray-400 dark:bg-gray-500"></span>
        <div className="flex gap-2">
          <div className="h-7 w-7 animate-pulse text-3xl text-gray-400"></div>
        </div>
      </div>

      <div className="flex items-center justify-between">
        <div className="flex items-center justify-center gap-2">
          <div className="h-24 w-24 animate-pulse rounded-full bg-gray-400 dark:bg-gray-500"></div>
          <span className="h-20 w-24 animate-pulse rounded bg-gray-400 text-6xl dark:bg-gray-500"></span>
        </div>
        <div className="flex items-center justify-center">
          <span className="h-7 w-20 animate-pulse rounded bg-gray-400 text-xl font-bold dark:bg-gray-500"></span>
        </div>
      </div>
    </div>
  </div>
);

export { WeatherCardSkeleton };
