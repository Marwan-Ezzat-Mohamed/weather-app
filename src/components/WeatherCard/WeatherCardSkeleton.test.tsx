import { render } from "@testing-library/react";
import { WeatherCardSkeleton } from "./WeatherCardSkeleton";

test("WeatherCardSkeleton matches snapshot", () => {
  const { container } = render(<WeatherCardSkeleton />);
  expect(container).toMatchSnapshot();
});
