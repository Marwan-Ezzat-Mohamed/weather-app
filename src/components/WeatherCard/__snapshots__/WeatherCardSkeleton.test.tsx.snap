// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardSkeleton matches snapshot 1`] = `
<div>
  <div
    class="min-w-sm relative flex w-full max-w-sm animate-pulse justify-center rounded-2xl bg-gray-300 p-5 dark:bg-gray-600"
  >
    <div
      class="w-full space-y-5"
    >
      <div
        class="flex justify-between"
      >
        <span
          class="h-7 w-2/3 animate-pulse rounded bg-gray-400 dark:bg-gray-500"
        />
        <div
          class="flex gap-2"
        >
          <div
            class="h-7 w-7 animate-pulse text-3xl text-gray-400"
          />
        </div>
      </div>
      <div
        class="flex items-center justify-between"
      >
        <div
          class="flex items-center justify-center gap-2"
        >
          <div
            class="h-24 w-24 animate-pulse rounded-full bg-gray-400 dark:bg-gray-500"
          />
          <h1
            class="h-20 w-24 animate-pulse rounded bg-gray-400 text-6xl dark:bg-gray-500"
          />
        </div>
        <div
          class="flex items-center justify-center"
        >
          <span
            class="h-7 w-20 animate-pulse rounded bg-gray-400 text-xl font-bold dark:bg-gray-500"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
